### PR TITLE
Make JavaScriptServices use ProjectReference

### PR DESCRIPTION
**Pull request by jkotalik**
Few notes:
- I had to add MVC as a package reference to some samples (required until MVC uses project ref)
- We may want to delete Microsoft.AspNetCore.NodeServices.Sockets as it hasn't shipped for quite a while

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -26,6 +26,8 @@
   <PropertyGroup Label="Package Versions: Pinned">
     <!-- Packages that came from aspnet/Extensions. -->
     <MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>2.1.1</MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>
+    <!-- Adding MVC here until it is moved to project ref -->
+    <MicrosoftAspNetCoreMvcPackageVersion>2.1.1</MicrosoftAspNetCoreMvcPackageVersion>
```
Comment (natemcmaster) [inline build/dependencies.props]: Can we do MVC first? Doing these out of order will break the master branch.

@ajaybhargavb - are you looking into converting MVC to project ref?

**Comment by natemcmaster** [comment]
I think the links in the README's need to be updated. I see some broken links in https://github.com/aspnet/AspNetCore/tree/jkotalik/jss21/src/Middleware/SpaServices and https://github.com/aspnet/AspNetCore/tree/jkotalik/jss21/src/Middleware/NodeServices

**Comment by natemcmaster** [comment]
Since we're removing NodeServices.Sockets as a part of this, let's remove mention of it from samples and READMEs, like this section:https://github.com/aspnet/AspNetCore/tree/jkotalik/jss21/src/Middleware/NodeServices#built-in-hosting-models

File: build/dependencies.props (hunk 1 above)
Comment (ajaybhargavb) [reply build/dependencies.props]: I haven't started. I can start on this today.

**Review by natemcmaster — APPROVED** [review]

File: src/Middleware/NodeServices/README.md [hunk 2]
```diff
@@ -37,7 +36,7 @@ For .NET Framework apps:
 In that case, you don't need to use NodeServices directly (or install it manually). You can either:
 
 * **Recommended:** Use the `aspnetcore-spa` Yeoman generator to get a ready-to-go starting point using your choice of client-side framework. [Instructions here.](http://blog.stevensanderson.com/2016/05/02/angular2-react-knockout-apps-on-aspnet-core/)
-* Or set up your ASP.NET Core and client-side Angular/React/KO/etc. app manually, and then use the [`Microsoft.AspNetCore.SpaServices`](https://github.com/aspnet/JavaScriptServices/tree/dev/src/Microsoft.AspNetCore.SpaServices) package to add features like server-side prerendering or Webpack middleware. But really, at least try using the `aspnetcore-spa` generator first.
+* Or set up your ASP.NET Core and client-side Angular/React/KO/etc. app manually, and then use the [`Microsoft.AspNetCore.SpaServices`](https://github.com/aspnet/aspnetcore/tree/release/2.1/src/Middleware/SpaServices/) package to add features like server-side prerendering or Webpack middleware. But really, at least try using the `aspnetcore-spa` generator first.
```
Comment (natemcmaster) [inline src/Middleware/NodeServices/README.md]:
```suggestion
* Or set up your ASP.NET Core and client-side Angular/React/KO/etc. app manually, and then use the [`Microsoft.AspNetCore.SpaServices`](/src/Middleware/SpaServices/) package to add features like server-side prerendering or Webpack middleware. But really, at least try using the `aspnetcore-spa` generator first.
```

FYI - you can make links relative to the repo root like this.

Comment (jkotalik) [reply src/Middleware/NodeServices/README.md]: 😮

**Pull request merged**